api/proto/metrics/metrics.proto

```diff
@@ -177,12 +177,32 @@ message appCpuMetric {
   uint64 total_ns = 7;                 // like 'total' but in nanoseconds. Overflow may happen after ~40 years without EVE restart
 }
 
-// StorageMetric   At the moment not used
+// StorageDeviceMetrics - contains device statistics from ZFS storage
+message StorageDeviceMetrics {
+  string name = 1;              // device name or pool name in storage
+  string serial_number = 2;     // empty unless it's a disk
+  uint64 alloc = 3;             // space allocated (in byte)
+  uint64 total = 4;             // total capacity (in byte)
+  uint64 deflated_space = 5;    // deflated capacity (in byte)
+  uint64 replaceable_size = 6;  // replaceable dev size (in byte)
+  uint64 expandable_size = 7;   // expandable dev size (in byte)
+  uint64 read_errors = 8;       // counter read count errors
+  uint64 write_errors = 9;      // counter write errors
+  uint64 checksum_errors = 10;  // counter checksum errors
+  uint64 bytes_read = 11;       // bytes read (in byte)
+  uint64 bytes_write = 12;      // bytes written (in byte)
+  uint64 ops_count_read = 13;   // Number of read ops
+  uint64 ops_count_write = 14;  // Number of write ops
+}
+
+// StorageMetric - contains pool statistics including child devices
 message StorageMetric {
   string pool_name = 1;
   // Depreated tag 2
   // Depreated tag 3
   // Depreated tag 4
+  StorageDeviceMetrics pool_stats = 5;     // Pool stats from ZFS
+  repeated StorageDeviceMetrics disks = 6; // disk stats in pool from ZFS
```

Comment:
When we have a hierachy of vdevs (e.g., a RAID-0/stripe where each member is a RAID-1/mirror) wouldn't want to show that hiearchy in the metrics as well?

Reply:
Such a task was not set before me, but if you think that we need it, I will add it.

Reply:
Sorry for the delay, but the metrics need to match the config/info hierachy that @giggsoff added.

Reply:
Corrected